src/utils/iqeEnablement.ts

```diff
@@ -103,10 +103,26 @@ function init(store: Store, libJwt?: () => LibJWT | undefined) {
    * Check response errors for cross_account requests.
    * If we get error response with specific cross account error message, we kick the user out of the corss account session.
    */
-  window.fetch = function fetchReplacement(path = '', options, ...rest) {
+  window.fetch = function fetchReplacement(path: URL | RequestInfo, options, ...rest) {
     const tid = Math.random().toString(36);
     const additionalHeaders: any = spreadAdditionalHeaders(options);
 
+    /**
+     * If path is a Request object, it may contain headers. Those headers will be overwritten, not merged,
+     * by the headers property containing the authorization headers and additionalHeaders in options. Therefore,
+     * we need to merge any headers present in the Request object into additionalHeaders.
+     */
+
+    function isRequest(path: URL | RequestInfo): path is Request {
+      return (path as Request).headers !== undefined;
```

Comment:
I'm still a bit new to Typescript and I'm not sure if this is the best way to implement the type guard but it seems to work.

Reply:
The `RequestInfo` type itself is defined as `type RequestInfo = Request | string`. We can use that in combination with the `instanceof` operator to get the correct type without any typecasting.

```ts
function isRequest(p: URL | RequestInfo): p is Request {
  return !(p instanceof URL) && typeof p !== 'string';
}
```

Your code will work, but this one is a bit safer in the event that for whatever reason someone assigns the `headers` property to the URL instance which is possible in JS. (I am sure there is a nicer way to write the condition)